cmd/minishift/cmd/start.go

```diff
@@ -83,68 +76,49 @@ var (
 	hostPvDirectory      = "/var/lib/minishift/openshift.local.pv"
 )
 
-// startFlagSet contains the minishift specific command line switches
-var startFlagSet = flag.NewFlagSet(commandName, flag.ContinueOnError)
-
 // clusterUpFlagSet contains the command line switches which needs to be passed on to 'cluster up'
-var clusterUpFlagSet = flag.NewFlagSet(commandName, flag.ContinueOnError)
-
-// subscription Manager FlagSet contains username and password details
-var subscriptionManagerFlagSet = flag.NewFlagSet(commandName, flag.ContinueOnError)
+var clusterUpFlagSet *flag.FlagSet
 
 // minishiftToClusterUp is a mapping between flag names used in minishift CLI and flag name as passed to 'cluster up'
 var minishiftToClusterUp = map[string]string{
 	"openshift-env":     "env",
 	"openshift-version": "version",
 }
 
-// runner executes commands on the host
-var runner util.Runner = &util.RealRunner{}
+// init configures the command line options of this command
+func init() {
+	clusterUpFlagSet = initClusterUpFlags()
+	startCmd.Flags().AddFlagSet(clusterUpFlagSet)
+	startCmd.Flags().AddFlagSet(initStartFlags())
+	startCmd.Flags().AddFlagSet(initSubscriptionManagerFlags())
+
+	viper.BindPFlags(startCmd.Flags())
+	RootCmd.AddCommand(startCmd)
 
-func SetRunner(newRunner util.Runner) {
-	runner = newRunner
+	provision.SetDetector(&provisioner.MinishiftProvisionerDetector{Delegate: provision.StandardDetector{}})
 }
 
-// runStart is executed as part of the start command
+// runStart handles all command line arguments, launches the VM and provisions OpenShift
 func runStart(cmd *cobra.Command, args []string) {
 	libMachineClient := libmachine.NewClient(constants.Minipath, constants.MakeMiniPath("certs"))
 	defer libMachineClient.Close()
 
+	ensureNotRunning(libMachineClient, constants.MachineName)
```

Comment:
It is part of the pre-flight checks we recently discussed. I noticed that if I had a machine running and did a 'minishift start', it would still do cluster up stuff. Effectively to no prevail. This is a check which will check whether the VM is running and if so, it will print a message of the form "Minishift VM already running".